src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/ChoosePaymentMethodHandler.php

```diff
@@ -73,8 +68,30 @@ public function __invoke(ChoosePaymentMethod $choosePaymentMethod): OrderInterfa
         $payment = $this->paymentRepository->findOneByOrderId($choosePaymentMethod->paymentId, $cart->getId());
         Assert::notNull($payment, 'Can not find payment with given identifier.');
 
+        if (in_array(
+            $cart->getCheckoutState(),
+            [OrderCheckoutStates::STATE_SHIPPING_SELECTED, OrderCheckoutStates::STATE_SHIPPING_SKIPPED, OrderCheckoutStates::STATE_PAYMENT_SELECTED],
+            true)
+        ) {
```

Comment:
This check seems wrong. Instead of extending the logic of `choosing payment method` we should create a separate handler, which will handle changing payment methods after checkout is completed.